src/main/java/life/qbic/expdesign/io/MHCLigandDesignReader.java

```diff
@@ -723,8 +751,15 @@ public List<TechnologyType> getTechnologyTypes() {
 
   @Override
   public Map<String, List<String>> getParsedCategoriesToValues(List<String> header) {
-    logger.warn("Method getParsedCategoriesToValues not implemented.");
-    return new HashMap<>();
+    Map<String, List<String>> res = new HashMap<>();
+    for (String cat : header) {
+      if (parsedCategoriesToValues.containsKey(cat)) {
+        res.put(cat, new ArrayList<>(parsedCategoriesToValues.get(cat)));
+      } else {
+        logger.warn(cat + " not found");
+      }
+    }
+    return res;
```

Comment:
The `Map<String, List<String>>` res makes it not entirely clear what is supposed to end up in this map. As this is found quite often, maybe introducing named types would be interesting.